usd/src/usd_parser/USDTransforms_TEST.cc

```diff
@@ -0,0 +1,195 @@
+/*
+ * Copyright (C) 2022 Open Source Robotics Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+*/
+
+#include <vector>
+
+#include <gtest/gtest.h>
+
+#include <ignition/utilities/ExtraTestMacros.hh>
+
+#include "test_config.h"
+#include "test_utils.hh"
+
+// TODO(ahcorde) this is to remove deprecated "warnings" in usd, these warnings
+// are reported using #pragma message so normal diagnostic flags cannot remove
+// them. This workaround requires this block to be used whenever usd is
+// included.
+#pragma push_macro ("__DEPRECATED")
+#undef __DEPRECATED
+#include <pxr/usd/usd/stage.h>
+#pragma pop_macro ("__DEPRECATED")
+
+#include "sdf/usd/usd_parser/USDTransforms.hh"
+
+void checkTransforms(
+  const std::string &_primName,
+  pxr::UsdStageRefPtr &_stage,
+  const ignition::math::Vector3d &_translation,
+  const std::vector<ignition::math::Quaterniond> &_rotation,
+  const ignition::math::Vector3d &_scale)
+{
+  pxr::UsdPrim prim = _stage->GetPrimAtPath(pxr::SdfPath(_primName));
+  ASSERT_TRUE(prim);
+
+  sdf::usd::UDSTransforms usdTransforms =
+    sdf::usd::ParseUSDTransform(prim);
+
+  EXPECT_EQ(_translation, usdTransforms.Translate());
+  EXPECT_EQ(_scale, usdTransforms.Scale());
+  EXPECT_EQ(_rotation.size(), usdTransforms.Rotations().size());
+  for (unsigned int i = 0; i < _rotation.size(); ++i)
```

Comment:
Can you add a check that the two vectors being compared here are of the same size to avoid out of bounds access errors?
```suggestion
  ASSERT_EQ(_rotation.size(), usdTransforms.Rotations().size());
  for (unsigned int i = 0; i < _rotation.size(); ++i)
```